extensions/replication/tasks/UpdateReplicationStatus.js

```diff
@@ -97,6 +97,97 @@ class UpdateReplicationStatus extends BackbeatTask {
+     * @param {string} site - site name.
+     * @param {Logger} log - Logger
+     * @return {undefined}
+     */
+    _pushReplayEntry(queueEntry, site, log) {
+        queueEntry.decReplayCount();
+        const count = queueEntry.getReplayCount();
+        const retryEntry = queueEntry.toRetryEntry(site).toKafkaEntry();
+        const topicName = this.replayTopics[count];
+        if (topicName && this.replayProducers[topicName]) {
+            this.replayProducers[topicName].publishReplayEntry(retryEntry);
+            log.info('push failed entry to the replay topic', {
+                entry: queueEntry.getLogInfo(),
+                topicName,
+                count,
+                site,
+            });
+        } else {
+            log.error('error pushing failed entry to the replay topic',
+                {
+                    entry: queueEntry.getLogInfo(),
+                    topicName,
+                    count,
+                    site,
+                });
+        }
+    }
+
+    /**
+     * Manage entry that failed replication.
+     * if replay topics are not defined in the configuration, replay logic is disabled.
+     * else:
+     *  - if count value has not been set,
+     *      - set it to max total attempt,
+     *      - decrement count value by 1,
+     *      - push entry to "replay topic".
+     *  - if count value > 0
+     *      - decrement count value by 1,
+     *      - push entry to "replay topic".
+     *  - if count value is 0
+     *      - push entry "failed topic"
+     *      - update source object md with site replication status set to FAILED
+     * @param {QueueEntry} refreshedEntry - Updated queue entry with latest "source object" metadata.
+     * @param {QueueEntry} queueEntry - The queue entry with the failed status.
+     * @param {string} site - site name.
+     * @param {Logger} log - Logger
+     * @return {null | ObjectQueueEntry} updated "source object" metadata - either null
+     * if no "source object" metadata update needed, or ObjectQueueEntry with the "source object" metadata
+     */
+    _handleFailedReplicationEntry(refreshedEntry, queueEntry, site, log) {
+        if (!this.replayTopics) {
+            // if replay topics are not defined in the configuration,
+            // replay logic is disabled.
+            if (this.repConfig.monitorReplicationFailures) {
+                this._pushFailedEntry(queueEntry);
+            }
+            return refreshedEntry.toFailedEntry(site);
+        }
+        const count = queueEntry.getReplayCount();
+        const totalAttemps = this.replayTopics.length;
+        if (count === 0) {
+            if (this.repConfig.monitorReplicationFailures) {
+                this._pushFailedEntry(queueEntry);
+            }
+            return refreshedEntry.toFailedEntry(site);
+        }
+        if (count > 0) {
+            if (count > totalAttemps) { // might happen if replay config has changed
```

Comment:
This is giving the entry "one last chance" if it has exceeded the new retry limit, shouldn't it be considered failed instead per the new replay policy?

Reply:
The check is there in case the entry has (for e.g.) 4 more retries (count === 4) and the `replayTopics` configuration gets changed to something like:
```
"replayTopics": [
    {
        "topicName": "backbeat-replication-replay-0",
        "retries": 2
    },
],
```
We want to reset the count to 2.